lib/validation.rb

```diff
@@ -208,4 +208,11 @@ def self.validate_short_text(text, field_name)
   def self.validate_account_name(name)
     fail ValidationFailed.new({name: "Name must only contain letters, numbers, spaces, and hyphens and have max length 63."}) unless name&.match(ALLOWED_ACCOUNT_NAME)
   end
+
+  def self.validate_url(url)
+    uri = URI.parse(url)
+    fail ValidationFailed.new({url: "Invalid URL"}) if !["http", "https"].include?(uri.scheme) || uri.host.nil? || uri.host.empty?
```

Comment:
Should we be more explicit if it doesn't start with "http/https"? It might be difficult for customers to understand that it's invalid due to the prefix.

Reply:
fair question: should the code support non-https? Should we let people complain and analyze their product problem to see if we should allow unencrypted communication?

Reply:
I changed this to only support https. If we get enough complaints, we can consider supporting self-signed certificates (by asking user to upload their public CA cert)